lib/offset_calculator.rb

```diff
@@ -1,33 +1,36 @@
+require_relative 'key_generator'
 require 'date'
 
 class OffsetCalculator
-  attr_reader :date
-  def initialize
-    @date = Time.now.strftime("%m%d%y")
-  end
+  attr_reader :kg,
+              :date,
+              :a_rot,
+              :b_rot,
+              :c_rot,
+              :d_rot
 
-  def date_squared(date)
-    @date.to_i ** 2
-  end
-  # should we rename our method so that we don't have a variable of the same name?
-  def offset
-    convert_offset = date_squared(date).to_s.chars
-    convert_offset[-4, 4]
+  def initialize(kg = KeyGenerator.new, date = Date.today)
+    @kg   = kg
+    @date = date
   end
 
-  def a_offset
-    offset[0]
+  def date_format
+    "#{@date.mday}#{@date.mon}#{@date.year.to_s[-2, 2]}"
   end
 
-  def b_offset
-    offset[1]
+  def date_squared
+    date_format.to_i ** 2
   end
-
-  def c_offset
-    offset[2]
+  # should we rename our method so that we don't have a variable of the same name?
+  def date_array
+    convert_offset = date_squared.to_s.chars
+    convert_offset[-4, 4]
   end
 
-  def d_offset
-    offset[3]
+  def offset
```

Comment:
Is there a way to do this with an enumerable? I'm not understanding this method, where are these instance variables used?